Engine/Results/BaseResultsHandler.cs

```diff
@@ -509,6 +524,90 @@ protected virtual void SampleBenchmark(DateTime time, decimal value)
             Sample("Benchmark", "Benchmark", 0, SeriesType.Line, time, value);
         }
 
+        /// <summary>
+        /// Sample drawdown of equity of the strategy
+        /// </summary>
+        /// <param name="time">Time of the sample</param>
+        /// <param name="currentPortfolioValue">Current equity value</param>
+        protected virtual void SampleDrawdown(DateTime time, decimal currentPortfolioValue)
+        {
+            // This will throw otherwise, in this case just don't sample
+            if (CumulativeMaxPortfolioValue != 0)
+            {
+                // Calculate our drawdown and sample it
+                var drawdown = Statistics.Statistics.DrawdownPercent(currentPortfolioValue, CumulativeMaxPortfolioValue);
+                Sample("Drawdown", "Equity Drawdown", 0, SeriesType.Line, time, drawdown, "%");
+            }
+        }
+
+        /// <summary>
+        /// Sample assets sales volume
+        /// </summary>
+        /// <param name="time">Time of the sample</param>
+        protected virtual void SampleSalesVolume(DateTime time)
+        {
+            var volumeBySymbol = Algorithm.Portfolio.Values
+                .Select(x => new KeyValuePair<Symbol, decimal>(x.Symbol, x.TotalSaleVolume));
```

Comment:
nit - We don't really need the select operation here?

Think we should probably skip if `TotalSaleVolume` is 0, thinking algorithms selecting many symbols or custom symbols